software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/riak/RiakNodeImpl.java

```diff
@@ -99,15 +108,51 @@ public boolean isPackageDownloadUrlProvided() {
 
     @Override
     protected Collection<Integer> getRequiredOpenPorts() {
-        // TODO this creates a huge list of inbound ports; much better to define on a security group using range syntax!
-        int erlangRangeStart = getConfig(ERLANG_PORT_RANGE_START).iterator().next();
-        int erlangRangeEnd = getConfig(ERLANG_PORT_RANGE_END).iterator().next();
+        Integer erlangRangeStart = config().get(ERLANG_PORT_RANGE_START);
+        Integer erlangRangeEnd = config().get(ERLANG_PORT_RANGE_END);
+        sensors().set(ERLANG_PORT_RANGE_START, erlangRangeStart);
+        sensors().set(ERLANG_PORT_RANGE_END, erlangRangeEnd);
+
+        boolean configureInternalNetworking = config().get(CONFIGURE_INTERNAL_NETWORKING);
```

Comment:
I'd do the `Boolean.TRUE.equals(config().get(CONFIGURE_INTERNAL_NETWORKING))`, in case of null.